.github/workflows/lint.yaml

```diff
@@ -13,43 +13,34 @@ jobs:
     env:
       BUNDLE_WITHOUT: ruby_shadow:omnibus_package
     steps:
-    - uses: actions/checkout@v2
+    - uses: actions/checkout@v3
     - uses: ruby/setup-ruby@v1
       with:
-        ruby-version: 2.7
+        ruby-version: 3.1
         bundler-cache: true
     - uses: r7kamura/rubocop-problem-matchers-action@v1 # this shows the failures in the PR
     - run: bundle exec chefstyle
 
   spellcheck:
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v2
-      - uses: carlosperate/download-file-action@v1.0.3
+      - uses: actions/checkout@v3
+      - uses: carlosperate/download-file-action@v2.0.0
         id: download-custom-dictionary
         with:
           file-url: 'https://raw.githubusercontent.com/chef/chef_dictionary/main/chef.txt'
           file-name: 'chef_dictionary.txt'
-      - uses: zwaldowski/cspell-action@v1
-        with:
-          config: cspell.json
-          paths: "**/*"
+      - uses: streetsidesoftware/cspell-action@v2.12.0
 
   coverage-test:
     name: Coverage
-    runs-on: ubuntu-18.04
+    runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v2
-      - name: Set up ruby 2.7
+      - uses: actions/checkout@v3
+      - name: Set up ruby 3.1
         uses: ruby/setup-ruby@v1
         with:
-          ruby-version: 2.7
+          ruby-version: 3.1
           bundler-cache: true
       - name: run specs
         run: bundle exec rake spec --trace
-      - name: Simplecov Report
```

Comment:
shall we restore the simplecov report run? any issue it appears to cause?

Reply:
It keeps failing causing the whole job to fail even though it passed the rspec tests. I also checked out other chef org repos and didn't see any similar jobs so removed it.

Reply:
i recalled this step fails for PRs from fork, so restored it